### PR TITLE
Allow admins to set initial password for OAuth-only users

### LINKED ISSUE #152: Allow admins to set an initial password for OAuth-only users
## Problem

`adminResetPasswordAction` in `lib/actions/user-actions.ts:288` only **updates** existing credential rows. For OAuth-only users (e.g., a Microsoft-OAuth signup with no email/password ever set), the action bails out at line 320-326 with "User does not have a password-based account".

That means admins currently have no self-service way to bootstrap a password for those users without the user themselves first signing in and going through the new self-service flow added in #151. There's a real workflow where admins want to provision a password for someone (onboarding, account recovery, support cases) without that round trip.

## Goal

The "Reset Password" form on `/dashboard/users/<id>/edit` should work for any non-deleted user — credential or OAuth-only. When no credential row exists, create one; when one exists, update it (current behavior).

## Plan

1. **`lib/actions/user-actions.ts:288`** — change `adminResetPasswordAction` to handle both cases:
   - If a `credential` account exists → update its password (current behavior; transaction with `session.deleteMany` to revoke active sessions, unchanged).
   - If no `credential` account exists → create one (`accountId === userId`, hashed password). No session revocation needed (no credential session exists yet, and the user's OAuth session is unaffected).
   - Wrap the create in try/catch on Prisma `P2002` so a race against the new self-service flow surfaces as a clean error rather than a 500.
2. **Activity logging** — currently `adminResetPasswordAction` doesn't call `logActivity` at all (the `PASSWORD_RESET` entry comes from better-auth's `/reset-password` endpoint, which the admin action doesn't hit). Add explicit `logActivity` calls so the audit trail distinguishes:
   - `PASSWORD_RESET` on the update branch.
   - `PASSWORD_SET` on the create branch.
   Both with `actorId = admin`, `targetType = "user"`, `targetId = subject user ID`.
3. **Tests** — add unit tests for the create branch (new credential), the update branch (existing credential), and the P2002 race.
4. **Form copy** — leave the page copy as "Reset Password" for now; conceptually the admin is just setting the password to whatever they typed. A follow-up could conditionally show "Set Password" when the user has no credential row.

## Dependency

Blocked on / based on #151 — uses the `@@unique([userId, providerId])` constraint added there for race safety, and uses the `PASSWORD_SET` enum value added there for the audit trail. The branch for this issue should be cut from `issue-150-oauth-set-initial-password` so it has access to both; once #151 merges, this PR will reconcile to a clean diff against `main`.

## Risks / notes

- No freshness gate needed — admins are gated by `requireRole("ADMIN")`; their permission to do this comes from being an admin, not recent re-auth.
- The current action's transaction (`account.update` + `session.deleteMany`) only applies to the update branch. Create branch is a single insert, no transaction needed.
- `canAssignRole` permission check is preserved unchanged.

### PR DESCRIPTION
**Pull request by chrisgen19**
Closes #152.

## Summary
- `adminResetPasswordAction` now handles OAuth-only users by **creating** a credential account when none exists, in addition to the existing **update** behaviour.
- Race-safe against the self-service flow added in #151: `P2002` on the `(user_id, provider_id)` unique surfaces as a clean "just set, reload" error rather than a 500.
- Adds explicit `logActivity` calls — `PASSWORD_RESET` on update, `PASSWORD_SET` on create — with `actorId = admin`, `targetType = "user"`, `targetId = subject`. Logging failures are caught so a successful persist is never reported as a failure (same pattern as `setInitialPasswordAction`).
- Update branch keeps its transactional session revocation; create branch is a single insert (no credential session exists yet, OAuth session is left untouched).
- `canAssignRole` permission gate and soft-delete guard are unchanged.

## Test plan
- [x] `bun run test` — 246/246 passing (8 new tests in `lib/actions/user-actions.test.ts`)
- [x] `bun run lint` — clean
- [x] `bunx tsc --noEmit` — clean
- [ ] Manual: as admin, open `/dashboard/users/<id>/edit` for an OAuth-only user, submit Reset Password → verify credential row created and `PASSWORD_SET` row in activity log
- [ ] Manual: as admin, repeat for a credential user → verify password updated, sessions revoked, `PASSWORD_RESET` row logged
- [ ] Manual: try to reset password for a SUPERADMIN as ADMIN → still rejected with permission error
- [ ] Manual: try to reset password for a soft-deleted user → still rejected

## Release Notes

* **Bug Fixes**
  * Admin password reset now handles all account states gracefully, including cases where credential accounts didn't previously exist.
  * Improved error handling for race conditions during password updates.
  * Enhanced activity logging with proper categorization of password reset versus initial setup actions.

* **Tests**
  * Added comprehensive test coverage for admin password reset operations, including edge cases and error scenarios.

### CODE REVIEW

File: lib/actions/user-actions.ts [hunk 1]
```diff
@@ -321,21 +323,62 @@ export async function adminResetPasswordAction(userId: string, formData: unknown
...
-		]);
+		if (account) {
+			await prisma.$transaction([
+				prisma.account.update({
+					where: { id: account.id },
+					data: { password: hashedPassword },
+				}),
+				prisma.session.deleteMany({
+					where: { userId },
+				}),
```
[the remark this answers is not in the document]
Comment (chrisgen19) [reply lib/actions/user-actions.ts:336]: Disagree — this is the **update** branch (existing credential user, real password reset). The issue explicitly specifies `session.deleteMany` here is "unchanged". The "leave OAuth sessions alone" guidance applies to the **create** branch (OAuth-only user getting an initial password), where this PR correctly skips revocation entirely.

Also, the suggested `provider: 'credentials'` / `authMethod: 'password'` filter isn't possible against our schema — better-auth's `Session` model has no provider/authMethod column; sessions are user-level, not per-account. See `prisma/schema.prisma`.

Resolving as not-an-issue.
